### PR TITLE
Feature/resource tracking

### PR DESCRIPTION
**Pull request by petersteneteg**
Fixes #...

Changes proposed in this PR:
 * ...
 * ...
 * ...

This has been tested on: ...

> Thank you for contributing to the Inviwo repository! Please ensure the following: 
> - Code is error and warning free
> - Code follows the Inviwo coding guidelines: 
>   https://github.com/inviwo/inviwo/wiki/Coding-Conventions
> - New Code / Processors follows the Inviwo documentation guidelines: 
>   https://github.com/inviwo/inviwo/wiki/Documentation-guide
> - New Code / processors are tested in a unittest and / or regression test: 
>   https://github.com/inviwo/inviwo/wiki/Manual#testing
> 
> Please remember to specify the environment this code been compiled and tested on:
> Operating System / Compiler / IDE / QT / CMake / Python versions.

### CODE REVIEW

File: include/inviwo/core/resourcemanager/resourcemanager.h [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (martinfalk) [inline include/inviwo/core/resourcemanager/resourcemanager.h]: spurious empty line?

File: modules/opengl/src/texture/texture2d.cpp [hunk 2]
```diff
@@ -117,6 +120,12 @@ void Texture2D::initialize(const void* data) {
                  static_cast<GLsizei>(dimensions_.y), 0, format_, dataType_, data);
     LGL_ERROR;
     forEachObserver([](TextureObserver* o) { o->notifyAfterTextureInitialization(); });
+
+    auto old = resource::remove(resource::GL{id_});
+    resource::add(resource::GL{id_}, Resource{.dims = glm::size4_t{dimensions_, 0, 0},
```
Comment (martinfalk) [inline modules/opengl/src/texture/texture2d.cpp:125]: won't this add the same resource multiple times if `initialize()` is called more than once?

Comment (martinfalk) [reply modules/opengl/src/texture/texture2d.cpp:125]: This applies to all textures.

File: src/core/ports/port.cpp [hunk 3]
```diff
@@ -41,7 +41,7 @@ Port::Port(std::string_view identifier, Document help)
 Processor* Port::getProcessor() const { return processor_; }
 
 std::string Port::getPath() const {
-    if (!processor_) throw Exception("No processor set", IVW_CONTEXT);
+    if (!processor_) return fmt::format("<not in any processor>.{}", identifier_);
```
Comment (martinfalk) [inline src/core/ports/port.cpp:44]: nice catch

**Review by martinfalk — DISMISSED** [review]
Neat feature. Waited long for something like this :)

File: modules/opengl/src/texture/texture2d.cpp (hunk 2 above)
Comment (petersteneteg) [reply modules/opengl/src/texture/texture2d.cpp:125]: If the id is already added it will just update/replace that one.

**Review by martinfalk — APPROVED** [review]

**Pull request merged**